src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -272,17 +271,17 @@ public object ResolveList(JsonElement array, IObjectField fieldSchema, ref IMeta
             }
             else if (array.ValueKind is JsonValueKind.String)
             {
-                using ArrayPoolWriter buffer = new();
-
-                string text = array.GetString()!;
-                int neededCapacity = Encoding.UTF8.GetMaxByteCount(text.Length);
-                int written = Encoding.UTF8.GetBytes(text, buffer.GetSpan(neededCapacity));
-                buffer.Advance(written);
-
-                Utf8JsonReader reader = new(buffer.GetWrittenSpan());
-                foreach (JsonElement element in JsonElement.ParseValue(ref reader).EnumerateArray())
+                string jsonString = array.GetString()!;
+                if (!string.IsNullOrEmpty(jsonString))
                 {
-                    resolvedList.Add(element);
+                    JsonDocument jsonDocument = JsonDocument.Parse(jsonString);
```

Comment:
When is this JsonDocument disposed off? The primary reason of not using JsonDocument was leaked memory..  Is that problem no longer applicable here?

Reply:
thanks, updated